lib/content-services/src/lib/tree/components/tree.component.spec.ts

```diff
@@ -0,0 +1,259 @@
+        fixture.detectChanges();
+        const treeHeaderDisplayName = fixture.nativeElement.querySelector(`[data-automation-id="tree-header-display-name"]`);
+        expect(treeHeaderDisplayName.innerText.trim()).toBe('test');
+    });
+
+    it('should show a list of nodes', () => {
+        component.refreshTree();
+        fixture.detectChanges();
+        const displayNameCellValueNode1 = getDisplayNameValue('testId1');
+        const displayNameCellValueNode2 = getDisplayNameValue('testId2');
+        expect(displayNameCellValueNode1).toBe('testName1');
+        expect(displayNameCellValueNode2).toBe('testName2');
+    });
+
+    it('should pad the tree according to the level of the node', () => {
+        component.treeService.treeNodes = Array.from(treeNodesChildrenMockExpanded);
+        fixture.detectChanges();
+        const nodeLevel0Padding = getNodePadding('testId1');
+        const nodeLevel0Padding2 = getNodePadding('testId2');
+        const nodeLevel1Padding = getNodePadding('testId3');
+        expect(nodeLevel0Padding).toEqual(nodeLevel0Padding2);
+        expect(nodeLevel1Padding).toBeGreaterThan(nodeLevel0Padding);
+    });
+
+    it('should show a spinner for nodes that are loading subnodes', () => {
+        component.treeService.treeNodes = Array.from(treeNodesChildrenMockExpanded);
+        fixture.detectChanges();
+        component.treeService.treeControl.dataNodes[0].isLoading = true;
+        fixture.detectChanges();
+        const nodeSpinner = getNodeSpinner('testId1');
+        expect(nodeSpinner).not.toBeNull();
+    });
+
+    it('should show a spinner while the tree is loading', () => {
+        fixture.detectChanges();
+        component.loadingRoot$ = of(true);
+        fixture.detectChanges();
+        const matSpinnerElement = fixture.nativeElement.querySelector('.adf-tree-loading-spinner-container .mat-progress-spinner');
+        expect(matSpinnerElement).not.toBeNull();
+    });
+
+    it('should show provided expand/collapse icons', () => {
+        component.treeService.treeNodes = Array.from(treeNodesMockExpanded);
+        component.expandIcon = 'folder';
+        component.collapseIcon = 'chevron_left';
+        component.treeService.collapseNode(component.treeService.treeNodes[0]);
+        fixture.detectChanges();
+        let nodeIcons: any = fixture.debugElement.queryAll(By.css('.adf-icon'));
+        expect(nodeIcons[0].nativeElement.innerText).toContain('folder');
+        spyOn(component.treeService.treeControl, 'isExpanded').and.returnValue(true);
+        fixture.detectChanges();
+        nodeIcons = fixture.debugElement.queryAll(By.css('.adf-icon'));
+        expect(nodeIcons[0].nativeElement.innerText).toContain('chevron_left');
+    });
+
+    it('should emit pagination when nodes are loaded', (done) => {
+        component.treeService.treeNodes = Array.from(treeNodesMockExpanded);
+        component.paginationChanged.subscribe((pagination: PaginationModel) => {
```

Comment:
if the "paginationChanged" is typed, you don't need to provide explicit typing in the "subscribe" as TS and IDEs resolve that without a problem; that also helps having less imports of type types, if the only case you use those is explicit typing

Reply:
Type removed